code/game/machinery/shuttle/shuttle_engine.dm

```diff
@@ -0,0 +1,135 @@
+//-----------------------------------------------
+//-------------Engine Thrusters------------------
+//-----------------------------------------------
+
+#define ENGINE_HEAT_TARGET 600
+
+/obj/machinery/shuttle/engine
+	name = "shuttle thruster"
+	desc = "A thruster for shuttles."
+	density = TRUE
+	obj_integrity = 250
+	max_integrity = 250
+	icon = 'icons/turf/shuttle.dmi'
+	icon_state = "burst_plasma"
+	idle_power_usage = 150
+	circuit = /obj/item/circuitboard/machine/shuttle/engine
+	var/thrust = 0
+	var/fuel_use = 0
+	var/bluespace_capable = TRUE
+	var/cooldown = 0
+	var/thruster_active = FALSE
+	var/obj/machinery/atmospherics/components/unary/shuttle/heater/attached_heater
```

Comment:
This should utilize weakrefs. You can find examples in the `code/modules/turbolift` folder or you can ping me on discord and I can explain how they work.

Reply:
Trying it